libs/openFrameworks/sound/ofAVEngineSoundPlayer.m

```diff
@@ -0,0 +1,709 @@
+//
+//  ofAVEngineSoundPlayer.cpp
+//  soundPlayerExample
+//
+//  Created by Theo Watson on 3/24/21.
+//
+
+#include "ofAVEngineSoundPlayer.h"
+#include "ofUtils.h"
+#include "ofMath.h"
+#include "ofLog.h"
+
+#define TARGET_OSX
+
+//REFS: https://github.com/ooper-shlab/AVAEMixerSample-Swift/blob/master/AVAEMixerSample/AudioEngine.m
+// https://developer.apple.com/documentation/avfaudio/avaudioengine
+// https://developer.apple.com/forums/thread/14138
+// https://developer.apple.com/forums/thread/48442
+// https://github.com/garynewby/XYAudioView/blob/master/XYAudioView/BasicAVAudioEngine.m
+// https://github.com/twilio/video-quickstart-ios/blob/master/AudioDeviceExample/AudioDevices/ExampleAVAudioEngineDevice.m
+
+#import <Foundation/Foundation.h>
+#import <AVFoundation/AVFoundation.h>
+#import <Accelerate/Accelerate.h>
+
+@interface AVEnginePlayer : NSObject
+
+@property(nonatomic, retain) NSTimer * timer;
+
+- (BOOL)loadWithFile:(NSString*)file;
+- (BOOL)loadWithPath:(NSString*)path;
+- (BOOL)loadWithURL:(NSURL*)url;
+- (BOOL)loadWithSoundFile:(AVAudioFile*)aSoundFile;
+
+- (void)unloadSound;
+
+- (void)play;
+- (void)play: (float)startTime;
+- (void)pause;
+- (void)stop;
+
+- (BOOL)isLoaded;
+- (BOOL)isPlaying;
+
+- (void)volume:(float)value;
+- (float)volume;
+
+- (void)pan:(float)value;
+- (float)pan;
+
+- (void)speed:(float)value;
+- (float)speed;
+
+- (void)loop:(BOOL)value;
+- (BOOL)loop;
+
+- (void)multiPlay:(BOOL)value;
+- (BOOL)multiPlay;
+
+- (void)position:(float)value;
+- (float)position;
+
+- (void)positionMs:(int)value;
+- (int)positionMs;
+
+- (float)positionSeconds;
+- (float)soundDurationSeconds;
+
+- (AVAudioFile *)getSoundFile;
+
+- (AVAudioEngine *)engine;
+
+@end
+
+
+@interface AVEnginePlayer ()
+
+//@property(nonatomic, strong) AVAudioEngine *engine;
+@property(nonatomic, strong) AVAudioMixerNode *mainMixer;
+@property(nonatomic, strong) AVAudioUnitVarispeed *variSpeed;
+@property(nonatomic, strong) AVAudioPlayerNode *soundPlayer;
+@property(nonatomic, strong) AVAudioFile *soundFile;
+@property(nonatomic, assign) bool mShouldLoop;
+@property(nonatomic, assign) int mGaurdCount;
+@property(nonatomic, assign) int mRestorePlayCount;
+@property(nonatomic, assign) bool mMultiPlay;
+@property(nonatomic, assign) float mRequestedPositonSeconds;
+@property(nonatomic, assign) AVAudioFramePosition startedSampleOffset;
+
+
+@end
+
+@implementation AVEnginePlayer
+
+@synthesize timer;
```

Comment:
basically, no need to explicit `@synthesize` if we use timer as `self.timer` (or `_timer` on internally. `_timer` will be defined automatically with `@prorpety` declaration).
if we want to use simply as `timer` then `@synthesize timer` is needed.